app/controllers/project_users.py

```diff
@@ -47,7 +47,48 @@ def post(self, project_id):
         user = User.find_first(email=validated_project_user_data.get('email', None))
 
         if not user:
-            return dict(status='fail', message='User not found'), 404
+            # register anonymous user
+            anonymous_user_email = validated_project_user_data.get('email', None)
+            anonymous_user_exists = AnonymousUser.find_first(email=anonymous_user_email)
+
+            if anonymous_user_exists:
+                return dict(status='fail', message='Annoymous user already exists'), 500
+
+            
+            role = validated_project_user_data.get('role', None)
+            if role == 'owner':
+                return dict(status='fail', message='User cannot be added as owner'), 400
+
+            new_anonymous_user = AnonymousUser(email=anonymous_user_email, project_id=project.id, role=role)
+            saved_anonymous_user = new_anonymous_user.save()
+
+            if not saved_anonymous_user:
+                return dict(status='fail', message='Internal Server Error'), 500
+            
+            # send anonymous user an invite email.
+            name = anonymous_user_email
+            template = "user/anonymous_user_to_project.html"
+            subject = "Assignment to Project from Crane Cloud: Invite To Register"
+            email =anonymous_user_email
+            today = date.today()
+            project_name = project.name
+            email_role = role
+
+            # send email
+            success = True
+            send_user_to_project_notification(
```

Comment:
I am kindly suggesting that you get a linter for your development environment, as some of your code is not well-linted.

Reply:
I have setup pylint